### DISCUSSION TITLE
How to build OKD sources?

### DISCUSSION THREAD

**Discussion by biruand1016** [okd-project/okd#1360] [Q&A]
Hello
I am trying to become contributor for okd community
So, I am learning OKD functions and develop workflow.
For testing I am gonna compile console operator
```
git clone https://github.com/openshift/console.git
git checkout -b release-4.11
podman build -t console-product:test -f Dockerfile_product
```
But I got errors
Error: error creating build container: initializing source docker://registry.ci.openshift.org/ocp/builder:rhel8-golang-1.18-openshift-4.11: reading manifest rhel8-golang-1.18-openshift-4.11 in [registry.ci.openshift.org/ocp/builder](http://registry.ci.openshift.org/ocp/builder): unauthorized: authentication required

Error: error creating build container: initializing source docker://registry.ci.openshift.org/ocp/builder:rhel8-base-nodejs-openshift-4.11: reading manifest rhel8-base-nodejs-openshift-4.11 in [registry.ci.openshift.org/ocp/builder](http://registry.ci.openshift.org/ocp/builder): unauthorized: authentication required

For fixing this issues, I changed registry.ci.openshift.org/ocp/builder:rhel8-golang-1.18-openshift-4.11 to registry.ci.openshift.org/openshift/release:golang-1.18. But I can't find correct image for nodejs
I can't access into [registry.ci.openshift.org](http://registry.ci.openshift.org/) and I can't see anything
How can I access this registry? Also how can I see what images are there in this registry?
How can I build okd sources?
Totally, I wanna know how to compile okd sources, and where is correct okd sources placed, and how to make build environment?
Regard

**Comment by vrutkovs** [comment]
See https://github.com/okd-project/okd-payload-pipeline

First you need [a centos image](https://github.com/okd-project/okd-payload-pipeline/blob/main/buildconfigs/02-additional-imagestream.yaml) you can use to build [`builder`](https://github.com/okd-project/okd-payload-pipeline/blob/main/buildconfigs/04-builder.yaml) and then the [`console`](https://github.com/okd-project/okd-payload-pipeline/blob/main/buildconfigs/10-console.yaml) itself
